app/src/main/java/org/thoughtcrime/securesms/onboarding/loading/LoadingViewModel.kt

```diff
@@ -23,25 +28,43 @@ import kotlin.time.Duration
 import kotlin.time.Duration.Companion.milliseconds
 import kotlin.time.Duration.Companion.seconds
 
-data class State(val duration: Duration)
+enum class State {
+    LOADING,
+    SUCCESS,
+    FAIL
+}
 
 private val ANIMATE_TO_DONE_TIME = 500.milliseconds
 private val IDLE_DONE_TIME = 1.seconds
 private val TIMEOUT_TIME = 15.seconds
 
-@OptIn(FlowPreview::class)
+private val REFRESH_TIME = 50.milliseconds
```

Comment:
This is an onboarding comment > @bemusementpark

Reply:
It's not in milliseconds, it's a `Duration` it's safer code, it's like type-safety, it could've saved the [Mars Climate Orbiter in '99](https://science.nasa.gov/mission/mars-climate-orbiter/)

```
public inline val Long.milliseconds get() = toDuration(DurationUnit.MILLISECONDS)
```